services/movie.py

```diff
@@ -0,0 +1,56 @@
+from typing import List
+
+from db.models import Actor, Genre, Movie
+
+
+def get_movies(genres_ids: List[int] = None,
+               actors_ids: List[int] = None) -> List[dict]:
+    # Start with all movies
+    movies = Movie.objects.all()
+
+    # Filter by genres if genres_ids are provided
+    if genres_ids:
+        movies = movies.filter(genres__id__in=genres_ids)
+
+    # Filter by actors if actors_ids are provided
+    if actors_ids:
+        movies = movies.filter(actors__id__in=actors_ids)
+
+    # Return unique movies (avoid duplicates due to many-to-many relationships)
+    return movies.distinct()
+
+
+def get_movie_by_id(movie_id: int) -> Movie:
```

Comment:
The function `get_movie_by_id` is annotated to return a `Movie`, but it returns `None` if the movie does not exist. Consider updating the return type annotation to `Optional[Movie]` to accurately reflect the possible return values.